### DISCUSSION TITLE
504 on self hosted instance

### DISCUSSION THREAD

**Discussion by noahstreller** [Infisical/infisical#2094] [Q&A]
Hi! 

I self host Infisical using Coolify, but for some reason, it returns a gateway timeout every now and then. I don't know how to reproduce it, it just happens randomly. I have not had this issue with any other service hosted on Coolify. Am I the only one? If you know anything about this problem, please let me know. 

Thanks!

**Comment by noahstreller** [comment]
I can't rely on Infisical if it has a 25% downtime; I'd really appreciate if someone knows anything about this issue.

![image](https://github.com/user-attachments/assets/13369394-e97e-4430-b0b2-f283c71aafe6)

**Comment by maidul98** [comment]
Hey @noahstreller , can you describe your deployment? Are you using nginx, HA proxy to route traffic to Infisical containers? How much CPU and memory are you giving it?

**Reply by noahstreller** [reply]
I have deployed Infisical using Coolify via Docker Compose and did not set any resource limits for it. The traffic is routed through a traefik proxy. There are no relevant logs during the 504.

**Reply by maidul98** [reply]
It is likely an issue with resource limits. Do you have any charts for memory and CPU usage?

**Reply by noahstreller** [reply]
Currently Sentinel metrics are disabled in Coolify due to some bugs. However, the 504 issue has miraculously resolved itself... No idea how it happened, there was no manual interaction with it. I'll keep an eye on it, and should it come up again, I'll update this thread. 

Thanks for your time 🙂
